native/c/zds.cpp

```diff
@@ -251,13 +251,19 @@ int zds_write_to_dsn(ZDS *zds, const string &dsn, string &data)
   {
     dsname = "//DD:" + string(zds->ddname);
   }
-  const string fopen_flags = zds->encoding_opts.data_type == eDataTypeBinary ? "wb" : "w" + string(",recfm=*");
+  // If file already exists, open in read+write mode to avoid losing ISPF stats
+  const string fopen_flags = zds->encoding_opts.data_type == eDataTypeBinary ? "rb+" : "r+" + string(",recfm=*");
+  const string fopen_flags2 = zds->encoding_opts.data_type == eDataTypeBinary ? "wb" : "w" + string(",recfm=*");
```

Comment:
(nitpick) The variable names are a bit vague, thoughts on making them more context-specific?